app/models/user.rb

```diff
@@ -87,6 +87,25 @@ def github_import=(import)
     @github_import = import
   end
 
+  %i(opted_in_newsletter opted_in_announcements opted_in_marketing_announcements opted_in_surveys opted_in_sponsorships opted_in_applications_open).each do |attribute|
+    define_method attribute.to_sym do
+      send("#{attribute}_at").present?
+    end
+
+    define_method "#{attribute}?".to_sym do
+      send(attribute)
+    end
+
+    define_method "#{attribute}=".to_sym do |value|
```

Comment:
Please also add model-level specs for this 🙏 
If you decide to put it into a concern, you could also put the specs into _"not-the-user-model-specs"_ so things stay not to polluted there either 😉